docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+3. Next.js uses the React Server Component Payload and JavaScript instructions to pre-render **HTML** on the server.
+
+> **What is React Server Component Payload?**
+>
+> The React Server Component Payload is a compact string representation of the React Component tree. It's used by React on the client to update the DOM. The React Server Component Payload contains:
+>
+> - The rendered result of Server Components
+> - Placeholders for where Client Components should be rendered on the client
+> - Any serialized props passed to these Client Components
+> - References to the JavaScript instructions needed to render the Client Components
+> - Metadata tags, such as references to stylesheets
+>
+> To learn more, see the [React Rendering](/docs/app/building-your-application/rendering/react-components) documentation.
+
+#### 2. Next.js Caching on the Server (Full Route Cache)
+
+On the server, at build time, Next.js caches the React Server Component Payload, Client Component JavaScript, and rendered HTML.
+
+#### 3. Next.js Caching on the Client (Router Cache)
+
+At request time, the React Server Component Payload is stored in the client-side [Router Cache](#router-cache) - a separate in-memory cache, split by individual routes. This Router Cache is used to improve the navigation experience by prefetching.
+
+#### 4. React Hydration and Reconciliation on the Client
+
+Then, on the client:
+
+1. The HTML is used to show an initial preview for Client and Server Components.
+2. The React Server Components Payload is used to reconcile the Client and rendered Server Component trees, and update the DOM.
+3. The JavaScript instructions are used to [hydrate](https://react.dev/reference/react-dom/client/hydrateRoot) Client Components.
+
+#### 5. Subsequent Navigations
+
+On subsequent navigations or during prefetching, Next.js will check if the React Server Components Payload is stored in Router Cache. If so, it will skip sending a new request to the server.
+
+If the route segments are not in the cache, Next.js will fetch the React Server Components Payload from the server, and populate the Router Cache on the client.
+
+#### Duration
+
+By default, the Full Route Cache is persistent. This means that the render output is cached across user requests.
+
+#### Invalidation
+
+There are two ways you can **invalidate** the Full Route Cache:
+
+- **Revalidating Data** This will invalidate the [Data Cache](#data-cache), which in turn will re-render components on the server, and cache the new render output.
+- **Redeploying**: Unlike the Data Cache, which persists across deployments, the Full Route Cache is cleared on new deployments.
+
+#### Opting out
+
+You can opt out of the Full Route Cache, or in other words, dynamically render components at request time, by:
+
+- **Using a [Dynamic Function](#dynamic-functions)**: This will dynamically render components at request time, but may reuse cached data, if any.
```

Comment:
```suggestion
- **Using a [Dynamic Function](#dynamic-functions)**: This will opt-out the route from the Full Route Cache. The Data Cache will still be used.
```